test/test-signal.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+// Testcase to produce NodeReport on signal interrupting a js busy-loop, 
+// showing it is interruptible.
+if (process.argv[2] === 'child') {
+  require('../').setEvents("signal");
+
+  // Exit on loss of parent process
+  process.on('disconnect', () => process.exit(2));
+
+  function busyLoop() {
+    var list = [];
+    for (var i=0; i<10000000000; i++) {
+      for (var j=0; i<1000; i++) {
+        list.push(new MyRecord());
+      }
+      for (var j=0; i<1000; i++) {
+        list[j].id += 1;
+        list[j].account += 2;
+      }
+      for (var j=0; i<1000; i++) {
+        list.pop();
+      }
+    }
+  }
+
+  function MyRecord() {
+    this.name = 'foo';
+    this.id = 128;
+    this.account = 98454324;
+  }
+
+  process.send('child started');
+  busyLoop();
```

Comment:
This should probably be `process.send('child started', busyLoop)`.  If for some reason the message can't be sent right away, it's going to get queued up but the queue will never be dispatched if you start busy looping immediately afterwards.